modules/scripts/startup-script/main.tf

```diff
@@ -26,6 +26,19 @@ locals {
     destination = "install_cloud_ops_agent_automatic.sh"
   }] : []
 
+  warnings = [
+    {
+      type        = "shell"
+      content     = "chmod -R o+r /var/log/journal"
+      destination = "access_joural.sh"
```

Comment:
typo: "journal"

Reply:
I'm not excited to change the permissions of an entire directory at boot, although I recognize this is a read-only change.

- it seems "wrong" because it impacts the domain of the system administrator
- a blind recursive chmod could take a long time to execute (would be better as a `find` command or determine very specific files we need to update)
- I haven't been convinced we've rejected alternatives

I'd like to see a very concise list of the situations we find this to be necessary. Probably some set of operating systems and OS Login configurations? If we're effectively solving a problem for 1 or 2 operating conditions, it may be simpler to consider -- for those situations -- a more generic warning (e.g. "startup scripts exist and you should be aware of them") that does not attempt to communicate the precise state of the system to the user.

Reply:
Will fix typo if we agree this is the way to move forward.

I understand the difficulty with the `chmod`, I don't like having to change permissions either.  That being said, there are a few different things at play:

1. I don't think its easy to determine which files to modify in the journal folder as they're created with what appear to be random suffixes (could get a list of the latest system.journal files or something)
2. The only alternative I can immediately think of, that maintains the error message, is to add users to the systemd-journal group, which is more difficult because users aren't created until login, or to use the `script-warning` file I created before.
   1. `script-warning` can be created by `passed-script.sh` file, which will get the exit code directly, and go around the need for users to need sudo, or to change permissions on any file
   2. The simplest alternative is to just ignore the error for users who log in after the scripts have finished.  The error will print for users who are logged in when the scripts finish.

We can discuss the rest offline.

Reply:
Additionally, we could also just add knobs to determine if the permissions for `/var/log/journal` will be given.  If it isn't the `profile.d` scripts won't be able to read anything and users who log in after the scripts will never see an error.